incubator/hnc/internal/reconcilers/hierarchy_config.go

```diff
@@ -316,11 +352,22 @@ func (r *HierarchyConfigReconciler) syncLabel(log logr.Logger, nsInst *corev1.Na
 	anc := ns
 	depth := 0
 	for anc != nil {
-		l := anc.Name() + labelDepthSuffix
+		l := anc.Name() + api.LabelTreeDepthSuffix
 		metadata.SetLabel(nsInst, l, strconv.Itoa(depth))
 		if anc.HasLocalCritCondition() {
 			break
 		}
+
+		// If the root is an external namespace, add all its external tree labels too.
+		// Note it's impossible to have an external namespace as a non-root, which is
+		// enforced by both admission controllers and the reconciler here.
+		if anc.IsExternal() && anc.Parent() == nil {
```

Comment:
So why are you checking for `anc.Parent()`? Just checking for `anc.IsExternal()` should be enough, and then you can `break` at the end of it.

Reply:
Ah right, done!